mtags/src/main/scala/scala/meta/internal/mtags/SymbolIndexBucket.scala

```diff
@@ -149,8 +149,14 @@ class SymbolIndexBucket(
       toplevel: String
   ): Unit = {
     if (source.isAmmoniteScript || !isTrivialToplevelSymbol(path, toplevel)) {
-      val acc = toplevels.getOrElse(toplevel, Set.empty)
-      toplevels(toplevel) = acc + source
+      val oldTopLevels = toplevels.getOrElse(toplevel, Set.empty)
+      // filter out no longer valid paths
+      val existingPaths = oldTopLevels.filter(_.exists) + source
+      toplevels(toplevel) = existingPaths + source
+
+      existingPaths.foreach { path =>
+        addMtagsSourceFile(path)
+      }
```

Comment:
@kpodsiad it might heavily affect the speed of indexing. 
Could you check the difference with this change ?

Reply:
> It'd be good to run some benchmarks, to see if 2) affects indexing. Probably we could separate addToplevelSymbol and updateToplevelSymbol and move logic which was added in this PR to the later method.

Yeah, I thought so. I think this should be refactor into 2 separate execution paths: initial indexing when there is no need to update entries and "update" case when we need to check entries. BTW, can you provide an example where one toplevel symbol can have more than one path? I just can't think of such case 🤔 Maybe we don't need set there?

Reply:
ok, I found https://github.com/scalameta/metals/pull/2668

Reply:
I quickly went over current benchmarks, but I think that currently there isn't any which tests toplevel indexing and then updates. I'm not familiar with this part of code though.

Reply:
Could we instead do it lazily within `query0` method? If one of the paths doesn't exist we could filter it out for query and update `definitions`/`toplevels`

Reply:
That will work. On my local branch I added another parameter to `addToplevelSymbol` method, but this needs to be passed through 4 different methods which is kind of meh. Your idea, on the other hand, is more elegant and I also like laziness, I'll give it a try.